scrambles/src/main/java/net/gnehzr/tnoodle/puzzle/SquareOnePuzzle.java

```diff
@@ -1,40 +1,45 @@
 package net.gnehzr.tnoodle.puzzle;
 
+import net.gnehzr.tnoodle.scrambles.*;
+import net.gnehzr.tnoodle.scrambles.AlgorithmBuilder.MergingMode;
 import net.gnehzr.tnoodle.svglite.Color;
 import net.gnehzr.tnoodle.svglite.Dimension;
 import net.gnehzr.tnoodle.svglite.Svg;
 import net.gnehzr.tnoodle.svglite.Transform;
 import net.gnehzr.tnoodle.svglite.Path;
 import net.gnehzr.tnoodle.svglite.Rectangle;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.Iterator;
-import java.util.Random;
-
-import net.gnehzr.tnoodle.scrambles.InvalidScrambleException;
-import net.gnehzr.tnoodle.scrambles.Puzzle;
-import net.gnehzr.tnoodle.scrambles.PuzzleStateAndGenerator;
+
+import java.util.*;
+
 import cs.sq12phase.FullCube;
 import cs.sq12phase.Search;
 import org.timepedia.exporter.client.Export;
 
 @Export
 public class SquareOnePuzzle extends Puzzle {
 
+    //public static final int SQ1_MAX_SCRAMBLE_LENGTH = 20;
+
     private static final int radius = 32;
 
+    private ThreadLocal<Search> twoPhaseSearcher;
+
     public SquareOnePuzzle() {
-        // TODO - we can't filter super aggresively until
-        // Chen Shuang's optimal solver is fixed.
-        //wcaMinScrambleDistance = 20;
         wcaMinScrambleDistance = 11;
+
+        twoPhaseSearcher = new ThreadLocal<Search>() {
+            protected Search initialValue() {
+                return new Search();
+            };
+        };
     }
 
     @Override
     public PuzzleStateAndGenerator generateRandomMoves(Random r) {
-        Search s = new Search();
-        String scramble = s.solution(FullCube.randomCube(r), Search.INVERSE_SOLUTION).trim();
+        FullCube randomState = FullCube.randomCube(r);
+
+        //String scramble = twoPhaseSearcher.get().solutionOpt(randomState, SQ1_MAX_SCRAMBLE_LENGTH, Search.INVERSE_SOLUTION).trim();
```

Comment:
Unsure which version to prefer -- `solution` certainly runs a bit faster than `solutionOpt`, and having sub-optimal scrambler sequences seems fine to me.

Please add your input on which version you prefer.